paasta_tools/config_utils.py

```diff
@@ -259,3 +261,82 @@ def commit_to_remote(self, extra_message: str = ""):
             _push_to_remote(self.branch)
         else:
             log.info("No files changed, no push required.")
+
+    def merge_recommendations(
+        self, recs: Dict[Tuple[str, str], Dict[str, Any]]
+    ) -> List[Dict[Tuple[str, str], Dict[str, Any]]]:
+        merged_recs = []
+        for (
+            service,
+            instance_type_cluster,
+        ), recommendations_by_instance in recs.items():
+            log.info(f"Starting to process {service}/{instance_type_cluster}.yaml...")
+            log.debug(
+                f"Getting current autotune configs for {service}/{instance_type_cluster}.yaml"
+            )
+            existing_recommendations = self.get_existing_configs(
+                service=service,
+                file_name=instance_type_cluster,
+                sub_dir=AUTO_SOACONFIG_SUBDIR,
+            )
+            log.debug(
+                f"Getting current configs for {service}/{instance_type_cluster}.yaml..."
+            )
+            existing_configs = self.get_existing_configs(
+                service=service,
+                file_name=instance_type_cluster,
+            )
+            for instance_name, recommendation in recommendations_by_instance.items():
+                log.debug(
+                    f"Merging recommendations for {instance_name} in {service}/{AUTO_SOACONFIG_SUBDIR}/{instance_type_cluster}.yaml..."
+                )
+                existing_recommendations.setdefault(instance_name, {})
+                existing_recommendations[instance_name].update(recommendation)
+                for limit_type, limits in (
+                    existing_configs[instance_name].get("autotune_limits", {}).items()
+                ):
```

Comment:
Would it make sense to move this whole for loop into its own function, maybe called `clamp_recommendations`? This function is a bit long.

Reply:
yup!